### PR TITLE
Fix endless diagnostic publishing when validation is disabled

### LINKED ISSUE #1175: Endless diagnostic publishing when validation is disabled
After updating from version 0.18.2 (As part of the vscode-xml) extension the `lemminx-linux` binary is using 100% of my cpu. (Same behaviour with 0.18.3, 0.18.4, 0.19.0). I don't have a specific xml file to point to, since it does so as soon as I open a container, without opening any XML files. I don't have debug symbols, but after attaching with GDB, these are the locations of all the threads: 
image

### PR DESCRIPTION
**Pull request by angelozerr**
Fix endless diagnostic publishing when validation is disabled

Fixes #1175

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLLanguageService.java [hunk 1]
```diff
@@ -168,6 +168,11 @@ public Hover doHover(DOMDocument xmlDocument, Position position, SharedSettings
 	public CompletableFuture<Path> publishDiagnostics(DOMDocument xmlDocument,
 			Consumer<PublishDiagnosticsParams> publishDiagnostics, Consumer<TextDocument> triggerValidation,
 			XMLValidationSettings validationSettings, CancelChecker cancelChecker) {
+		if (validationSettings != null && !validationSettings.isEnabled()) {
+			// Validation is disabled
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLLanguageService.java:172]: For me when validation is disabled, it should not report diagnostics.

[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLLanguageService.java]: Ok now I report empty diagnostics even if validation is disabled. We could improve this behavior in the future (track the change settings of enabled)

**Comment by rgrunber** [comment]
This fixes the issue for me. I don't see the excessive empty diagnostics published and the LS responds to changes of the validation setting.

**Comment by angelozerr** [comment]
> This fixes the issue for me. I don't see the excessive empty diagnostics published and the LS responds to changes of the validation setting.

Thanks for the review @rgrunber ! I'm waiting for @fbricon feedback.

**Comment by fbricon** [comment]
Thanks @angelozerr

**Pull request merged**